oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -157,14 +158,18 @@ private PropertySource oktaRedirectUriPropertySource(Environment environment) {
     }
 
     private PropertySource oktaStaticDiscoveryPropertySource(Environment environment) {
+        String oauth2Suffix = Optional.ofNullable(environment.getProperty("okta.oauth2.issuer"))
+            .filter(issuer -> issuer.contains("/oauth2"))
+            .map(s -> "")
+            .orElse("/oauth2");
```

Comment:
Nice clean little block!  It took a sec for my brain to grok it, maybe add a comment?

```suggestion
        String oauth2Suffix = Optional.ofNullable(environment.getProperty("okta.oauth2.issuer"))
            .filter(issuer -> issuer.contains("/oauth2")) // Check if URL is a Custom Authorization Server
            .map(s -> "") // if so, leave URL as is
            .orElse("/oauth2"); // for the Okta Org Authorization Server, /oauth2 needs to be appended
```

my suggestion might be a bit too verbose